tests/admin/test_controller.py

```diff
@@ -227,6 +227,86 @@ def test_details(self):
             eq_(1, len(unsuppress_links))
             assert lp.identifier.identifier in unsuppress_links[0]
 
+    def test_roles(self):
+        roles = self.manager.admin_work_controller.roles()
+        assert Contributor.ILLUSTRATOR_ROLE in roles.values()
+        assert Contributor.NARRATOR_ROLE in roles.values()
+        eq_(Contributor.ILLUSTRATOR_ROLE,
+            roles[Contributor.MARC_ROLE_CODES[Contributor.ILLUSTRATOR_ROLE]])
+        eq_(Contributor.NARRATOR_ROLE,
+            roles[Contributor.MARC_ROLE_CODES[Contributor.NARRATOR_ROLE]])
+
+    def test_languages(self):
+        languages = self.manager.admin_work_controller.languages()
+        assert 'en' in languages.keys()
+        assert 'fre' in languages.keys()
+        names = [name for sublist in languages.values() for name in sublist]
+        assert 'English' in names
+        assert 'French' in names
+
+    def test_media(self):
+        media = self.manager.admin_work_controller.media()
+        assert Edition.BOOK_MEDIUM in media.values()
+        assert Edition.medium_to_additional_type[Edition.BOOK_MEDIUM] in media.keys()
+
+    def _make_test_edit_request(self, data):
+        [lp] = self.english_1.license_pools
+        with self.request_context_with_library("/"):
+            flask.request.form = ImmutableMultiDict(data)
+            return self.manager.admin_work_controller.edit(
+                lp.identifier.type, lp.identifier.identifier
+            )
+
+    def test_edit_unknown_role(self):
+        response = self._make_test_edit_request(
+            [('contributor-role', self._str),
+             ('contributor-name', self._str)])
+        eq_(400, response.status_code)
+        eq_(UNKNOWN_ROLE.uri, response.uri)
+
+    def test_edit_invalid_series_position(self):
+        response = self._make_test_edit_request(
+            [('series', self._str),
+             ('series_position', 'five')])
+        eq_(400, response.status_code)
+        eq_(INVALID_SERIES_POSITION.uri, response.uri)
+
+    def test_edit_unknown_medium(self):
+        response = self._make_test_edit_request(
+            [('medium', self._str)])
+        eq_(400, response.status_code)
+        eq_(UNKNOWN_MEDIUM.uri, response.uri)
+
+    def test_edit_unknown_language(self):
+        response = self._make_test_edit_request(
+            [('language', self._str)])
+        eq_(400, response.status_code)
+        eq_(UNKNOWN_LANGUAGE.uri, response.uri)
+
+    def test_edit_invalid_date_format(self):
+        response = self._make_test_edit_request(
+            [('issued', self._str)])
+        eq_(400, response.status_code)
+        eq_(INVALID_DATE_FORMAT.uri, response.uri)
+
+    def test_edit_invalid_rating_not_number(self):
+        response = self._make_test_edit_request(
+            [('rating', self._str)])
```

Comment:
self._str returns a string version of a number; you might be getting this status code because the 'number' is too large. Recommend you change this to an obvious non-number string like "abc".